### ISSUE TITLE
Ambiguous snippets generated for list operations with a string parameter

### ISSUE THREAD

**Issue by jskeet** [googleapis/gapic-generator-csharp#198]
Currently trying to migrate [Dialogflow v2](https://github.com/googleapis/googleapis/tree/master/google/cloud/dialogflow/v2) to the microgenerator, and I'm running into issues with `ListIntents`. Proto definition:

```proto
// Returns the list of all intents in the specified agent.
rpc ListIntents(ListIntentsRequest) returns (ListIntentsResponse) {
  option (google.api.http) = {
    get: "/v2/{parent=projects/*/agent}/intents"
  };
  option (google.api.method_signature) = "parent";
  option (google.api.method_signature) = "parent,language_code";
}
...
message ListIntentsRequest {
  // Required. The agent to list all intents from.
  // Format: `projects/<Project ID>/agent`.
  string parent = 1 [
    (google.api.field_behavior) = REQUIRED,
    (google.api.resource_reference) = {
      type: "dialogflow.googleapis.com/Agent"
    }
  ];
  ...
}
```

Generated synchronous method signatures:

```csharp
ListIntents(ListIntentsRequest request, gaxgrpc::CallSettings callSettings = null)
ListIntents(string parent, string pageToken = null, int? pageSize = null, gaxgrpc::CallSettings callSettings = null)
ListIntents(string parent, string languageCode, string pageToken = null, int? pageSize = null, gaxgrpc::CallSettings callSettings = null)
ListIntents(AgentName parent, string pageToken = null, int? pageSize = null, gaxgrpc::CallSettings callSettings = null)
ListIntents(AgentName parent, string languageCode, string pageToken = null, int? pageSize = null, gaxgrpc::CallSettings callSettings = null)
```

Generated snippet code:

```csharp
AgentName parent = AgentName.FromProject("[PROJECT]");
string languageCode = "";
// Make the request
PagedAsyncEnumerable<ListIntentsResponse, Intent> response = intentsClient.ListIntentsAsync(parent, languageCode);
```

It's ambiguous between the last two methods.

In the monolithic generator, we used to put argument names for the equivalent snippet:

```csharp
PagedEnumerable<ListIntentsResponse, Intent> response =
    intentsClient.ListIntents(parent: parent, languageCode: languageCode);
```

**Comment by jskeet** [comment]
(I'll have a go at fixing this myself...)

**Comment by jskeet** [comment]
This is the relevant code in the monolithic generator: https://github.com/googleapis/gapic-generator/blob/cd919793e439d2d21505900aa08ef9f99519d8fa/src/main/java/com/google/api/codegen/transformer/csharp/CSharpGapicSnippetsTransformer.java#L150

**Comment by jskeet** [comment]
Ooh, it looks like there's already a test for this - https://github.com/googleapis/gapic-generator-csharp/blob/3af28067d03b494cdb462acd1f1000822b13aef1/Google.Api.Generator.Tests/ProtoTests/Paginated/Paginated.proto#L12
So I need to work out what's different about this situation, add a test for it, and fix it.

**Comment by jskeet** [comment]
Fortunately *most* of the logic is already fine - it's only overloads which use resource names which are broken. Test change is huge... hopefully production change will be small.

**Comment by chrisdunelm** [comment]
Closing, fixed in #199
